src/components/DatePicker/DatePicker.vue

```diff
@@ -0,0 +1,305 @@
+      >
+        Today
+      </a>
+    </div>
+
+    <transition :name="monthDirection == 1 ? 'slide' : 'slide-reverse'">
+      <div class="days-grid" :key="month">
+        <div class="days-header-grid">
+          <div
+            v-for="(n, i) in 7"
+            class="day-header"
+            :class="'day-' + getDayOfTheWeek(i)"
+            :key="i"
+          >
+            {{ getDisplayDay(i) }}
+          </div>
+        </div>
+
+        <button
+          v-for="day in Array.from({ length: getDayOfTheWeek(1) })
+            .map((e, i) => daysInPreviousMonth - i)
+            .reverse()"
+          :key="day"
+          class="day previous-month"
+          :class="{
+            ['day-' + getDayOfTheWeekPreviousMonth(day)]: true,
+            today: isToday(day, previousMonth, previousMonthYear)
+          }"
+          @click="selectDate(day, previousMonth, previousMonthYear)"
+        >
+          {{ day }}
+        </button>
+
+        <button
+          v-for="day in daysInMonth"
+          :key="day"
+          class="day"
+          :class="{
+            ['day-' + getDayOfTheWeek(day)]: true,
+            today: isToday(day, month, year),
+            'selected-day': isSelectedDay(day, month, year)
+          }"
+          @click="selectDate(day, month, year)"
+        >
+          {{ day }}
+        </button>
+
+        <button
+          v-for="day in 6 - getDayOfTheWeek(daysInMonth)"
+          :key="day"
+          class="day next-month"
+          :class="{
+            ['day-' + getDayOfTheWeekNextMonth(day)]: true,
+            today: isToday(day, nextMonth, nextMonthYear)
+          }"
+          @click="selectDate(day, nextMonth, nextMonthYear)"
+        >
+          {{ day }}
+        </button>
+      </div>
+    </transition>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Options, prop } from 'vue-class-component'
+
+class Props {
+  /**
+   * A reactive native JS DateTime Object (only the date portion is reactive)
+   * @model
+   */
+  modelValue = prop<Date>({ required: false })
+  /**
+   * A native JS DateTime Object used to set the default value for the component
+   */
+  value = prop<Date>({ required: false })
+}
+
+const Component = Options
+@Component({
+  watch: {
+    value_(val) {
+      this.$emit('update:modelValue', val)
+    },
+    modelValue(val) {
+      this.date = new Date(val)
+    },
+    value(val) {
+      this.date = new Date(val)
+    }
+  }
+})
+
+/**
+ * The DatePicker is meant to present a unified cross-browser date picking component, replacing native browser date pickers
+ * which vary drastically browser to browser.
+ *
+ * @displayName DatePicker
+ */
+export default class DatePicker extends Vue.with(Props) {
+  value_: Date = this.modelValue
+    ? new Date(this.modelValue)
+    : this.value
+    ? new Date(this.value)
+    : new Date()
+
+  date: Date = new Date(this.value_)
+  monthDirection: number = 0
+
+  get today(): Date {
+    return new Date()
+  }
```

Comment:
This will never change so it doesn't need to be computed. 

```suggestion
  today = new Date()
```